### DISCUSSION TITLE
Running flutter pub outdated is not possible when using local package via pubspec_override

### DISCUSSION THREAD

**Discussion by artahc** [invertase/melos#385] [Q&A] [answered]
Hi, 
I have several local packages, which depend on each other via generated `pubspec_override`. Most of them have common dependencies, such as mockito.
I wanted to check & sync these dependencies, but when I tried to run `melos exec --depends-on=mockito flutter pub outdated`, I got a message `Package not available (could not find package <my_local_package> at https://pub.dartlang.org).`

How do you do this operation? or is this operation is not supported?
Thank you

**Comment by blaugold** [comment]
This should work and I'm not able to reproduce this issue.
Could you provide the full error message and what version of Flutter you are using (`flutter --version`)?

**Reply by artahc** [reply]
Flutter 3.3.1 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 4f9d92fbbd (4 weeks ago) • 2022-09-06 17:54:53 -0700
Engine • revision 3efdf03e73
Tools • Dart 2.18.0 • DevTools 2.15.0

I only got these errors
`Package not available (could not find package baito_widgets at https://pub.dartlang.org).`
`pub finished with exit code 69`

In my local package pubspec, i use `mypackage: any` to reference my other local package, I think it has something to do with this

Edit:
I'm using melos v2.6.0

**Comment by fnicastri** [comment]
any news on this?
it's cumbersome 
@blaugold

**Comment by thipokch** [comment, marked as answer]
@blaugold It's happening to me too. It's probably because of this issue https://github.com/dart-lang/pub/issues/3502
```
    • Flutter version 3.3.10 on channel stable at /opt/homebrew/Caskroom/flutter/3.3.10/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 135454af32 (5 weeks ago), 2022-12-15 07:36:55 -0800
    • Engine revision 3316dd8728
    • Dart version 2.18.6
    • DevTools version 2.15.0
```

**Reply by thipokch** [reply]
@artahc as a workaround... I found that listing the path itself, instead of pubspec_override works, but defeats the purpose of Melos in the first place.

```yaml
mypackage:
    path: ../../path/to/package
```

**Reply by blaugold** [reply]
@artahc Thanks for posting your workaround! `any` is nice and easy for dependencies on local packages, but does not work with `dart pub outdate` for packages that have never been published, as explained in the issue you mentioned. For now, if `dart pub outdate` needs to be used in this situation, `path` dependencies have to be used for all local packages.

**Reply by fnicastri** [reply]
@blaugold 
and then why Melos? This soft of defeat half the scope of this utility.  
Maybe the solution could be to generate the path during the bootstrap?
